Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/NodeFunctionGeneric.h

```diff
@@ -80,10 +80,14 @@ namespace ScriptCanvas
         using ResultType = FunctionTraits::result_type;\
         static const size_t s_numArgs = FunctionTraits::arity;\
         static const size_t s_numNames = SCRIPT_CANVAS_FUNCTION_VAR_ARGS(__VA_ARGS__);\
-        /*static const size_t s_numResults = ScriptCanvas::Internal::extended_tuple_size<ResultType>::value;*/\
+        static const size_t s_numResults = ScriptCanvas::Internal::extended_tuple_size<ResultType>::value;\
         \
         static AZStd::string GetArgName(size_t i)\
         {\
+            if (s_numArgs < 2)\
```

Comment:
please help me to refresh memory, 2 means including execution in and out slot?

Reply:
in this case its the number of data inputs on the Node, which correspond to the number of arguments to the C++ function.

Reply:
oh yes it is for c++ function not scriptcanvas node.
btw, just to confirm, argument list won't have this pointer as first arg, right? I remember I saw this kind of thing somewhere before.

Reply:
Correct, that's for behavior context method nodes written against class methods.